app/services/topic_services.ml

```diff
@@ -294,17 +332,38 @@ let answer =
       let open Lwt_util in
       let open Models.Message in
       let*? message = handle_form request validate_creation in
-      let*? message_id, topic =
-        Dream.sql request @@ create user topic_id message
-      in
-      let+? () =
-        Env.get request
-        @@ Slack_services.new_answer user topic_id topic message_id
-      in
-      topic_id, message_id)
-    ~succeed:(fun (topic_id, message_id) request ->
-      Flash_info.action request "Message enregistré";
-      redirect_to ~anchor:message_id ~:Endpoints.Topic.show topic_id request)
+      if is_created_preview message
+      then
+        let*? topic, messages = get_full_topic request topic_id in
+        let preview_content = created_message message in
+        let current_time = Ptime_clock.now () in
+        let message_preview =
+          Models.Message.make ~id:"" ~content:preview_content user current_time
+        in
+        let messages = messages @ [ message_preview ] in
+        (* FIXME: "" *)
+        (* FIXME? this isn't great style. A better way would be module
+         * and inner types *)
```

Comment:
I'm not sure about the issue here.

Reply:
I don't really like tuples with lots of arguments and would have preferred two records. I changed it a bit still.